### PR TITLE
Add func/method comments to datastore

### PR DESCRIPTION
**Pull request by c3y1huang**
Added comments to exported functions and methods for datastore:
* datastore.go
* kubernetes.go
* longhorn.go
* uncached.go

### CODE REVIEW

File: datastore/kubernetes.go [hunk 1]
```diff
@@ -233,53 +266,74 @@ func (s *DataStore) ListInstanceManagerPods() ([]*corev1.Pod, error) {
...
 	return s.kubeClient.CoreV1().PersistentVolumes().Create(pv)
 }
 
+// DeletePersisentVolume deletes persistentvolumefor the given
+// persistentvolume name
 func (s *DataStore) DeletePersisentVolume(pvName string) error {
 	return s.kubeClient.CoreV1().PersistentVolumes().Delete(pvName, &metav1.DeleteOptions{})
 }
 
+// UpdatePersisentVolume updates the persistentvolume for the given
```
Comment (yasker) [inline datastore/kubernetes.go:286]: `PersistentVolume`. Same below.

File: datastore/longhorn.go [hunk 2]
```diff
@@ -60,11 +63,13 @@ func (s *DataStore) InitSettings() error {
 	return nil
 }
 
+// CreateSetting create longhorn setting for the given setting and namespace
```
Comment (yasker) [inline datastore/longhorn.go:66]: `Longhorn`, same below.

File: datastore/longhorn.go [hunk 3]
```diff
@@ -283,6 +297,7 @@ func GetOwnerReferencesForVolume(v *longhorn.Volume) []metav1.OwnerReference {
 	}
 }
 
+// CreateVolume create longhorn volume resource and verifies creation
```
Comment (yasker) [inline datastore/longhorn.go:300]: `creates a Longhorn Volume resource`.

**Comment by yasker** [comment]
Add our doc writer @catherineluse

File: datastore/kubernetes.go [hunk 4]
```diff
@@ -233,53 +266,74 @@ func (s *DataStore) ListInstanceManagerPods() ([]*corev1.Pod, error) {
...
 	return s.knLister.Get(name)
 }
 
+// CreatePersisentVolume creates persistentvolume for the given
+// persistentvolume object
 func (s *DataStore) CreatePersisentVolume(pv *corev1.PersistentVolume) (*corev1.PersistentVolume, error) {
 	return s.kubeClient.CoreV1().PersistentVolumes().Create(pv)
 }
 
+// DeletePersisentVolume deletes persistentvolumefor the given
```
[the remark this answers is not in the document]
Comment (c3y1huang) [reply datastore/kubernetes.go]: done

File: datastore/kubernetes.go [hunk 5]
```diff
@@ -233,53 +266,74 @@ func (s *DataStore) ListInstanceManagerPods() ([]*corev1.Pod, error) {
 	return res, nil
 }
 
+// GetKubernetesNode gets the node from the index for the given name
 func (s *DataStore) GetKubernetesNode(name string) (*corev1.Node, error) {
 	return s.knLister.Get(name)
 }
 
+// CreatePersisentVolume creates persistentvolume for the given
```
[the remark this answers is not in the document]
Comment (c3y1huang) [reply datastore/kubernetes.go]: done

File: datastore/kubernetes.go (hunk 1 above)
Comment (c3y1huang) [reply datastore/kubernetes.go]: done

File: datastore/longhorn.go (hunk 2 above)
Comment (c3y1huang) [reply datastore/longhorn.go]: done

File: datastore/longhorn.go (hunk 3 above)
Comment (c3y1huang) [reply datastore/longhorn.go]: done.

File: datastore/longhorn.go [hunk 6]
```diff
@@ -1358,6 +1438,7 @@ func (s *DataStore) UpdateInstanceManager(im *longhorn.InstanceManager) (*longho
 	return obj, nil
 }
 
+// UpdateInstanceManagerStatus updates Longhorn InstanceManager resource and verifies update
```
Comment (shuo-wu) [inline datastore/longhorn.go:1441]:
```
// UpdateInstanceManagerStatus updates Longhorn InstanceManager status resource and verifies update
```

**Review by shuo-wu — CHANGES_REQUESTED** [review]

File: datastore/uncached.go [hunk 7]
```diff
@@ -6,57 +6,119 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
-// Event list is uncached but current lister doesn't field selector, so we have no choice but read from API directly
-
+// GetLonghornEventList returns an uncached list of longhorn events for the given
+// namespace directly from the API server.
+// Using cached informers should be preferred but current lister doesn't field selector.
```
Comment (yasker) [inline datastore/uncached.go:11]: `doesn't have a field selector`. Same below.

**Comment by c3y1huang** [comment]
> LGTM, except for the commit message still contains 3 signed-off-by.

I've cleaned up the commit message. PTAL.

**Pull request merged**